### DISCUSSION TITLE
I cannot get enable_terrain_style to work for me

### DISCUSSION THREAD

**Discussion by wassimj** [pyvista/pyvista#3277] [Q&A]
Hello,
I have an app that uses pyvista to visualise 3D building models. I need to use enable_terrain_style to keep the building vertical. I cannot get it to work. It continues to orbit freely. The code is at:
[https://github.com/wassimj/TopologicVTK/blob/main/app.py](https://github.com/wassimj/TopologicVTK/blob/main/app.py)
Please look at the function named: def **pyvista_streamlit(plotter):**
Thank you for your help.

**Comment by banesullivan** [comment]
This is likely not working because you are exporting the scene to a web-based viewer:

https://github.com/wassimj/TopologicVTK/blob/86830eedee3478e12e10122e66a2da86010b3d63/app.py#L124

In this case, with `panel`. I'm not sure if the `panel`-based viewer has support for terrain style. You may want to try the `pythreejs` one instead

**Reply by arili0803** [reply]
Hello Mr. banesullivan, I would like to know that why I try to use plotter.enable_terrain_style with vs code but it still not work. Is it necessary to use jupyter? And how can I combine the add_mesh with panel or pythreejs? thanks!
